src/server/lib/middleware/csrf.ts

```diff
@@ -0,0 +1,29 @@
+import csrf from 'csurf';
+import { Express, NextFunction, Request } from 'express';
+import { ResponseWithLocals } from '@/server/models/Express';
+import { getConfiguration } from '@/server/lib/configuration';
+
+const isHttps = getConfiguration().isHttps;
+
+const csrfMiddleware = csrf({
+  cookie: {
+    key: '_csrf',
+    sameSite: true,
+    secure: isHttps,
+    httpOnly: true,
+  },
+});
+
+const csrfLocalsMiddleware = (
+  req: Request,
+  res: ResponseWithLocals,
+  next: NextFunction,
+) => {
+  res.locals.csrfToken = req.csrfToken();
+  next();
+};
+
+export const addCsrfMiddleware = (server: Express) => {
+  server.use(csrfMiddleware);
+  server.use(csrfLocalsMiddleware);
+};
```

Comment:
middleware can just be an array, so don't really need a function to do this

from the express [docs](https://expressjs.com/en/guide/using-middleware.html):

```js
function logOriginalUrl (req, res, next) {
  console.log('Request URL:', req.originalUrl)
  next()
}

function logMethod (req, res, next) {
  console.log('Request Type:', req.method)
  next()
}

var logStuff = [logOriginalUrl, logMethod]
app.get('/user/:id', logStuff, function (req, res, next) {
  res.send('User Info')
})
```

Reply:
done